File: ShoeStore/ShoeStore.Domain/Entities/Product.cs
namespace ShoeStore.Domain.Entities
{
    public class Product
    {
        public int Id { get; set; }
        public string? SKU { get; set; }
        public string Name { get; set; } = null!;
        public int? BrandId { get; set; }
        public int? SupplierId { get; set; }
        public decimal CostPrice { get; set; }
        public decimal OriginalPrice { get; set; }
        public string? Color { get; set; }
        public string? Size { get; set; }
        public string? Description { get; set; }
        public string? ImageUrl { get; set; }
        public int StatusId { get; set; }
        public DateTime CreatedAt { get; set; }

        // Navigation
        public Brand? Brand { get; set; }
        public Supplier? Supplier { get; set; }
        public Status Status { get; set; } = null!;

        public ICollection<StoreProduct>? StoreProducts { get; set; }
        public ICollection<OrderDetail>? OrderDetails { get; set; }
    }
}

File: shoestorefe/lib/domain/entities/product.dart
import 'store_quantity.dart';

class Product {
  final int id;
  final String? sku;
  final String name;
  final int? brandId;
  final int? supplierId;
  final double costPrice;
  final double originalPrice;
  final String? color;
  final String? size;
  final String? description;
  final String? imageUrl;
  final int statusId;
  final DateTime createdAt;
  final List<StoreQuantity> stores;

  Product({
    required this.id,
    this.sku,
    required this.name,
    this.brandId,
    this.supplierId,
    required this.costPrice,
    required this.originalPrice,
    this.color,
    this.size,
    this.description,
    this.imageUrl,
    required this.statusId,
    required this.createdAt,
    this.stores = const [],
  });
}
